Bottom: 52c3498efeb997fb321727c72f7774fcb7116e46
Top:    1a1a06cf7a3888ff2ad8409753312623a0b4de9b
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-06-27 01:36:58 +0100

line.c: Cleanup group checking in drc_lines()

Avoid using an out-of-bounds layer number which might be used in future.


---

diff --git a/src/line.c b/src/line.c
index 47e3125..f20f705 100644
--- a/src/line.c
+++ b/src/line.c
@@ -228,7 +228,8 @@ AdjustTwoLine (bool way)
 struct drc_info
 {
   LineType *line;
-  bool solder;
+  bool bottom_side;
+  bool top_side;
   jmp_buf env;
 };
 
@@ -249,7 +250,7 @@ drcPad_callback (const BoxType * b, void *cl)
   PadType *pad = (PadType *) b;
   struct drc_info *i = (struct drc_info *) cl;
 
-  if (TEST_FLAG (ONSOLDERFLAG, pad) == i->solder &&
+  if (TEST_FLAG (ONSOLDERFLAG, pad) == i->bottom_side &&
       !TEST_FLAG (FOUNDFLAG, pad) && LinePadIntersect (i->line, pad))
     longjmp (i->env, 1);
   return 1;
@@ -294,7 +295,7 @@ drc_lines (PointType *end, bool way)
   Coord dx, dy, temp, last, length;
   Coord temp2, last2, length2;
   LineType line1, line2;
-  Cardinal group, comp;
+  Cardinal group;
   struct drc_info info;
   bool two_lines, x_is_long, blocker;
   PointType ans;
@@ -321,14 +322,10 @@ drc_lines (PointType *end, bool way)
       length = abs (dy);
     }
   group = GetGroupOfLayer (INDEXOFCURRENT);
-  comp = max_group + 10;	/* this out-of-range group might save a call */
-  if (GetLayerGroupNumberByNumber (solder_silk_layer) == group)
-    info.solder = true;
-  else
-    {
-      info.solder = false;
-      comp = GetLayerGroupNumberByNumber (component_silk_layer);
-    }
+
+  info.bottom_side = (GetLayerGroupNumberBySide (BOTTOM_SIDE) == group);
+  info.top_side = (GetLayerGroupNumberBySide (TOP_SIDE) == group);
+
   temp = length;
   /* assume the worst */
   best = 0.0;
@@ -411,7 +408,7 @@ drc_lines (PointType *end, bool way)
 			drcVia_callback, &info);
 	      r_search (PCB->Data->pin_tree, &line1.BoundingBox, NULL,
 			drcVia_callback, &info);
-	      if (info.solder || comp == group)
+	      if (info.bottom_side || info.top_side)
 		r_search (PCB->Data->pad_tree, &line1.BoundingBox, NULL,
 			  drcPad_callback, &info);
 	      if (two_lines)
@@ -421,7 +418,7 @@ drc_lines (PointType *end, bool way)
 			    drcVia_callback, &info);
 		  r_search (PCB->Data->pin_tree, &line2.BoundingBox, NULL,
 			    drcVia_callback, &info);
-		  if (info.solder || comp == group)
+		  if (info.bottom_side || info.top_side)
 		    r_search (PCB->Data->pad_tree, &line2.BoundingBox, NULL,
 			      drcPad_callback, &info);
 		}
